addons/pos_self_order/models/product_product.py

```diff
@@ -100,10 +116,20 @@ def _get_product_for_ui(self, pos_config):
                 "pos_combo_ids": self.combo_ids.mapped("id") or False,
                 "is_pos_groupable": self.uom_id.is_pos_groupable,
                 "write_date": self.write_date.timestamp(),
+                "self_order_available": self.self_order_available,
             }
 
     def _get_self_order_data(self, pos_config: PosConfig) -> List[Dict]:
         return [
             product._get_product_for_ui(pos_config)
             for product in self
         ]
+
+    @api.constrains('self_order_available')
+    def _constrains_self_order_available(self):
+        config_self = self.env['pos.config'].search(['|', ('self_order_table_mode', '=', True), '|', ('self_order_view_mode', '=', True), ('self_order_kiosk', '=', True)])
+        for config in config_self:
+            if config.current_session_id and config.access_token:
+                self.env['bus.bus']._sendone(f'pos_config-{config.access_token}', 'PRODUCT_CHANGED', {
+                    'product': self._get_product_for_ui(config)
```

Comment:
Every time something changes, you'll send all the product information again?

Wouldn't it be easier to send the product ID with the new status?

What's more, if you resend the entire product each time, you'll have to reinstantiate its class each time.